### PR TITLE
[doc] Add some documentation on Sirius Web

### PR DESCRIPTION
**Pull request by sbegaudeau**
Signed-off-by: Stéphane Bégaudeau <stephane.begaudeau@obeo.fr>

# Pull request template

## General purpose
What is the main goal of this pull request?
- [ ] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [ ] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [ ] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [ ] Have the relevant issues been added to the pull request?
- [ ] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [ ] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: CONTRIBUTING [hunk 1]
```diff
@@ -1,36 +1,91 @@
...
+These branches are named using the following naming convention:
+
+```
+three_letters_identifying_the_contributor/kind_of_work/description
+```
+
+You could thus find in our repository branches such as:
+
+- `sbe/fix/diagram-related-issue`
+- `pcd/enh/new-feature-the-the-view-dsl`
```
Comment (pcdavid) [inline CONTRIBUTING]: `the-the`

File: CONTRIBUTING [hunk 2]
```diff
@@ -1,36 +1,91 @@
...
+For the few cases where a commit is not directly related to an issue, use a tag like [releng], [doc] or [cleanup]...
+Look for previously used tag before creating a new one.
+
+## Changelog
+
+All contributions should be mentionned in the file `CHANGELOG.adoc`.
+If a long description is necessary to provide additional details for a commit, it is necessary to provide those details in the changelog.
+Most of our commit messages do not contain a long description since those details are written in the changelog.
+
+Contributions which have a visual impact should try to provide screenshots showing the changes.
```
Comment (pcdavid) [inline CONTRIBUTING:68]: These images referenced from the changelog should be added in `doc/images`, not in the top-level directory.

File: doc/reference/01_overview.adoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+Sirius Web is an open source platform used to define modeling environments for your business concerns.
+With Sirius Web, you can define the concepts of your domain and some dedicated tooling.
+On the other hand, Sirius Components is a set of backend and frontend components which can be reused in various applications to build a modeling workbench based on representations.
+
+== Overview
+
+Sirius Web provides a modeling workbench to build model based applications.
+There are two kinds of users which will interact with Sirius Web, each one working with a specific aspect of the application.
+Firstly, specifiers will use Sirius Web to define all the concepts of their business and the graphical tooling to manipulate them.
```
Comment (pcdavid) [inline doc/reference/01_overview.adoc]: specifiers => studio makers

File: doc/reference/01_overview.adoc [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+=== View
+
+In a view, specifiers will create the descriptions of the representations they want to provide to their end users.
+The view can be used to describe various representations such as diagrams or forms to edit the data.
+
+Sirius Web comes with the `View` language to create and parameterize the various representations.
+In Sirius desktop, specifiers would use an `odesign` file to create the descriptions of the representations.
+It is possible to import those `odesign` files in Sirius Web to reuse some tooling created in Sirius desktop.
+
+Using a view, a specifier can create the description of a diagram which will be used by the end users to define various kinds of vehicules.
```
Comment (pcdavid) [inline doc/reference/01_overview.adoc]: Typo (here and in other instances below): `vehicles` (no `u`)

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: CONTRIBUTING (hunk 1 above)
Comment (sbegaudeau) [reply CONTRIBUTING]: Done

File: CONTRIBUTING (hunk 2 above)
Comment (sbegaudeau) [reply CONTRIBUTING:68]: Done

File: doc/reference/01_overview.adoc (hunk 3 above)
Comment (sbegaudeau) [reply doc/reference/01_overview.adoc]: Done

File: doc/reference/01_overview.adoc (hunk 4 above)
Comment (sbegaudeau) [reply doc/reference/01_overview.adoc]: I've fixed that and other typos

**Review by pcdavid — APPROVED** [review]

**Pull request merged**